deep_quoridor/src/quoridor_env.py

```diff
@@ -146,35 +165,48 @@ def _get_action_mask(self, agent_id):
             if not (0 <= new_row < self.board_size and 0 <= new_col < self.board_size):
                 continue
 
-            # Is there a wall in the way we're going?
-            blocked_by_wall = False
-            if row == 0:  # Moving horizontally - check vertical walls
-                wall_col = min(old_col, new_col)
-                # NOTE: The min and max tricks there are just a lazy way to avoid overindexing
-                # Since the wall grid is one smaller than the board (because boards take two spaces)
-                if (
-                    self.walls[min(new_row, self.wall_size - 1), wall_col, 0] == 1
-                    or self.walls[max(new_row - 1, 0), wall_col, 0] == 1
-                ):
-                    blocked_by_wall = True
-            else:  # Moving vertically - check horizontal walls
-                wall_row = min(old_row, new_row)
-                # NOTE: The min and max tricks there are just a lazy way to avoid overindexing
-                # Since the wall grid is one smaller than the board (because boards take two spaces)
-                if (
-                    self.walls[wall_row, min(new_col, self.wall_size - 1), 1] == 1
-                    or self.walls[wall_row, max(new_col - 1, 0), 1] == 1
-                ):
-                    blocked_by_wall = True
-
-            # Is the opponent in the target position?
-            opponent_in_target = self.positions[self._opponent(agent_id)] == (new_row, new_col)
+            # Check if that direction is blocked by a wall
+            if self._is_wall_between(old_row, old_col, new_row, new_col):
+                continue
 
-            # TODO: Jumping the opponent, considering walls if there are any
+            # Check if the opponent is in the target position
+            if self.positions[self._opponent(agent_id)] == (new_row, new_col):
+                # Can we jump straight over the component?
```

Comment:
I assume you mean the opponent?

```suggestion
                # Can we jump straight over the opponent?
```